### PR TITLE
Updates components-contrib with new io.Closer impl

### PR DESCRIPTION
**Pull request by JoshVanL**

### CODE REVIEW

**Review by elena-kolevska — APPROVED** [review]
LGTM assuming all tests pass after the components contrib PR is merged and go.mod is updated.

**Comment by yaron2** [comment]
Linter is failing

**Review by cicoyle — DISMISSED** [review]
Once linter is fixed I think this is good to merge

**Review by yaron2 — APPROVED** [review]

**Pull request merged**
